app/moves/middleware/set-results.moves.js

```diff
@@ -14,6 +14,12 @@ function setResultsMoves(bodyKey, locationKey) {
       const args = omitBy(req.body[bodyKey], isUndefined)
       const activeMoves = await req.services.move.getActive(args)
 
+      for (let i = 0; i < activeMoves.length; i++) {
+        const move = activeMoves[i]
+        const eventsForMove = await req.services.move.getByIdWithEvents(move.id)
+        move.timeline_events = eventsForMove.timeline_events
+      }
```

Comment:
```suggestion
      activeMoves.forEach((move) => {
        const eventsForMove = await req.services.move.getByIdWithEvents(move.id)
        move.timeline_events = eventsForMove.timeline_events
      })
```

I'm a little concerned about the performance implications of this though, it means we'd have to make an API call for every move, which [can lead to N+1 problems](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping) where the site gets slower and slower with the more moves you need to render.

Is there a way we could change the API to include this information already when you request the list of moves? I'm thinking either the `timeline_events` could be included directly, or if that in itself is quite expensive, could we add a field which marks whether the move has a `MoveLockout` event or not? In this way the API is doing the query against the database directly and should be more performant.

Reply:
Yeah I fully agree, would be good so when we call `req.services.move.getActive(args)` that it also returns the `timeline_events` with it.

Reply:
For visibility we have gone with the approach of sending over the `is_lockout` attribute from the api in this PR: https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/1795